server/src/main/java/org/opensearch/cluster/service/ClusterService.java

```diff
@@ -291,6 +291,16 @@ public final String getNodeName() {
         return nodeName;
     }
 
+    /**
+     * Functionality for register task type for cluster manager throttling.
+     *
+     * @param throttlingKey task key which needs to be registered for throttling.
+     * @param retryableOnDataNode has data retry mechanism for throttling exception
+     */
+    public void registerThrottlingKey(String throttlingKey, boolean retryableOnDataNode) {
+        clusterManagerService.registerThrottlingKey(throttlingKey, retryableOnDataNode);
+    }
+
```

Comment:
I think all the indirection will go away now

Reply:
https://github.com/opensearch-project/OpenSearch/pull/4542#discussion_r986828995